### PR TITLE
feat(api-rest): support making external promise cancellable

### PR DESCRIPTION
**Pull request by AllanZhengYP**
#### Description of changes

Support cancalling the external promise calling to internal `post` API,with new `updateReqeustToBeCancellable()` API. You can cancel your external promise like below:
```
import {
	post,
	cancel as cancelREST,
	updateRequestToBeCancellable,
} from '@aws-amplify/api-rest/internals';

const abortController = new AbortController();
const responsePromise = asyncFunctionCallingInternalPost();
updateRequestToBeCancellable(
	responsePromise,
	abortController
);
return responsePromise;

// cancel:
cancel(responsePromise, message);
```

#### Description of how you validated changes
Unit test, manaul test

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by david-mcafee — APPROVED** [review]

File: packages/api-rest/src/utils/createCancellableOperation.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	abortController: AbortController
+): Promise<HttpResponse>;
+/**
+ * Create a cancellable operation conforming to the external REST API interface.
+ * @internal
+ */
+export function createCancellableOperation(
+	handler: (signal: AbortSignal) => Promise<HttpResponse>
+): Promise<HttpResponse>;
+
```
Comment (elorzafe) [inline packages/api-rest/src/utils/createCancellableOperation.ts:24]: Why do you need the overload?

Comment (AllanZhengYP) [reply packages/api-rest/src/utils/createCancellableOperation.ts:24]: @elorzafe The other overload is for external REST APIs that will come in following PR.

File: packages/api-rest/src/common/internalPost.ts [hunk 2]
```diff
@@ -5,47 +5,66 @@ import { AmplifyClassV6 } from '@aws-amplify/core';
...
 export const cancel = (
 	promise: Promise<RestApiResponse>,
 	message?: string
 ): boolean => {
-	const cancelFn = cancelTokenMap.get(promise);
-	if (cancelFn) {
-		cancelFn(message);
+	const controller = cancelTokenMap.get(promise);
+	if (controller) {
+		controller.abort(message);
```
Comment (elorzafe) [inline packages/api-rest/src/common/internalPost.ts:51]: Does this work on react-native?

**Review by elorzafe — APPROVED** [review]
Looks good!

Thanks @AllanZhengYP 🏅

**Pull request merged**